torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -213,20 +210,46 @@ def parallelize_llama(model, world_mesh, parallel_dims, job_config: JobConfig):
 
         logger.info("Applied Tensor Parallelism to the model")
 
+    # apply AC + torch.compile
+    ac_config = job_config.activation_checkpoint
+    enable_compile = job_config.training.compile
+    for layer_id, transformer_block in enumerate(model.layers):
+        if ac_config.mode in ("full", "selective"):
+            transformer_block = checkpoint_wrapper(transformer_block, ac_config)
+        if enable_compile:
+            # turn on per-transformer block compile after AC wrapping and before FSDP
+            # TODO: dynamic shape have some issues so we turn it off for now.
+            transformer_block = torch.compile(transformer_block, dynamic=False)
```

Comment:
Curious how far we are from being able to enable `fullgraph=True`?

Reply:
I think `fullgraph=True` should work already when we moving to per-TransformerBlock compile, maybe I can add that flag too.

I ended up just turn dynamic=False in the current full mode compile https://github.com/pytorch/torchtitan/pull/297 in this case we can't full_graph=True yet as FSDP still graph breaking, but for that case we should also captured each TransformerBlock as full graphs already